crates/ruff_linter/src/rules/ruff/rules/missing_fstring_syntax.rs

```diff
@@ -0,0 +1,103 @@
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{
+    Arguments, Expr, ExprCall, Keyword, Stmt, StmtExpr, StringLiteral, StringLiteralValue,
+};
+use ruff_python_semantic::SemanticModel;
+use ruff_source_file::Locator;
+
+use super::helpers::should_be_fstring;
+
+/// # What it does
+/// Checks for strings that contain f-string syntax but are not f-strings.
+///
+/// ## Why is this bad?
+/// An f-string missing an `f` at the beginning won't format anything, and instead
+/// treat the interpolation syntax as literal.
+///
+/// ## Example
+///
+/// ```python
+/// name = "Sarah"
+/// dayofweek = "Tuesday"
+/// msg = "Hello {name}! It is {dayofweek} today!"
+/// ```
+///
+/// It should instead be:
+/// ```python
+/// name = "Sarah"
+/// dayofweek = "Tuesday"
+/// msg = f"Hello {name}! It is {dayofweek} today!"
+/// ```
+#[violation]
+pub struct MissingFStringSyntax {
+    literal: String,
+}
+
+impl AlwaysFixableViolation for MissingFStringSyntax {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let Self { literal } = &self;
+        format!(r#"`{literal}` may be a formatting string without an `f` prefix."#)
+    }
+
+    fn fix_title(&self) -> String {
+        "Add an `f` prefix to the formatting string".into()
+    }
+}
+
+pub(crate) fn missing_fstring_syntax(
+    diagnostics: &mut Vec<Diagnostic>,
+    value: &StringLiteralValue,
+    locator: &Locator,
+    semantic: &SemanticModel,
+) {
+    if let Stmt::Expr(StmtExpr { value, .. }) = semantic.current_statement() {
```

Comment:
Another option: check if `semantic.current_expression_parent().is_none()`, I think? Since that implies that this is a top-level expression.

Reply:
As per the test cases, I believe this is checking if it's a standalone string literal and abort if so.

If that's the case and we decide to add support for string literals that are implicitly concatenated to a f-string, we might need to update this part to include `Expr::FString` as well.